### PR TITLE
Numeric quiz

### LINKED ISSUE #38: Numerical Quiz
Numerical Questions to be implemented with tolerance values.

### PR DESCRIPTION
**Pull request by thesantatitan**
### Add Numeric Quiz in the types of Quizzes

- This fixes #38 
- Earlier alphanumeric quizzes were incorrectly characterized as numeric quizzes in a lot of places in the code, this was fixed.
- A field 'errorRate' was added to the KBC node in database, and functions in Databases/KBC.js functions were modified to accomodate this
- Autograder and mailingSystem in the cloud functions were changed to also use numeric quizzes

### CODE REVIEW

**Review by HackGod2000 — APPROVED** [review]

**Review by anuneetanand — APPROVED** [review]

File: App/Components/Quiz/QuizResultGraph.js [hunk 1]
```diff
@@ -41,8 +41,8 @@ export default class QuizResultGraph extends Component {
                     this.props.QuizMailer()
                 }
             }
-            else if(this.props.quizType==='numerical' || this.props.quizType=="multicorrect"){
-                await kbcResponse.getAllNumericalResponse(this.props.passCode, r["startTime"], r["endTime"] )
+            else if(this.props.quizType==='alphaNumerical' || this.props.quizType=="multicorrect" || this.props.quizType=="numeric"){
+                await kbcResponse.getAllalphaNumericalResponse(this.props.passCode, r["startTime"], r["endTime"] )
```
Comment (Sanchit-Trivedi) [inline App/Components/Quiz/QuizResultGraph.js:45]: TypeError: kbcResponse.getAllalphaNumericalResponse is not a function. (In 'kbcResponse.getAllalphaNumericalResponse(_this.props.passCode, r["startTime"], r["endTime"])', 'kbcResponse.getAllalphaNumericalResponse' is undefined)

The kbcResponse.getAllalphaNumericalResponse function is not defined anywhere in the codebase.

Comment (vishwesh-D-kumar) [reply App/Components/Quiz/QuizResultGraph.js:45]: @shreeya13garg Camel Case the function properly also when you define it

Comment (thesantatitan) [reply App/Components/Quiz/QuizResultGraph.js]: I fixed this error and the camel case problem in e318167

**Comment by vishwesh-D-kumar** [comment]
@shreeya13garg Overwriting for quizzes faculty side needs to be updated

**Comment by vishwesh-D-kumar** [comment]
@thesantatitan index.js will have to be refactored again to support the new implementation . Resolve all merge conflicts. 
@anuneetanand @HackGod2000 we will need to test cloud-functions after this,along with working functionality


@shreeya13garg a small demo video with the PR would be great.

**Review by vishwesh-D-kumar — APPROVED** [review]

**Comment by vishwesh-D-kumar** [comment]
Good work @shreeya13garg @thesantatitan ! 🎉

**Review by HackGod2000 — APPROVED** [review]

**Pull request merged**